plugin/src/main/cpp/extensions/openxr_fb_space_warp_extension_wrapper.cpp

```diff
@@ -0,0 +1,289 @@
+	ERR_FAIL_COND_MSG(singleton != nullptr, "An OpenXRFbSpaceWarpExtensionWrapper singleton already exists.");
+
+	request_extensions[XR_FB_SPACE_WARP_EXTENSION_NAME] = &fb_space_warp_ext;
+	singleton = this;
+}
+
+OpenXRFbSpaceWarpExtensionWrapper::~OpenXRFbSpaceWarpExtensionWrapper() {
+	cleanup();
+	singleton = nullptr;
+}
+
+godot::Dictionary OpenXRFbSpaceWarpExtensionWrapper::_get_requested_extensions() {
+	godot::Dictionary result;
+	for (auto ext : request_extensions) {
+		godot::String key = ext.first;
+		uint64_t value = reinterpret_cast<uint64_t>(ext.second);
+		result[key] = (godot::Variant)value;
+	}
+	return result;
+}
+
+uint64_t OpenXRFbSpaceWarpExtensionWrapper::_set_system_properties_and_get_next_pointer(void *p_next_pointer) {
+	system_space_warp_properties.next = p_next_pointer;
+	return reinterpret_cast<uint64_t>(&system_space_warp_properties);
+}
+
+uint64_t OpenXRFbSpaceWarpExtensionWrapper::_set_projection_views_and_get_next_pointer(int p_view_index, void *p_next_pointer) {
+	if (fb_space_warp_ext && enabled) {
+		space_warp_info[p_view_index].next = p_next_pointer;
+		return reinterpret_cast<uint64_t>(&space_warp_info[p_view_index]);
+	} else {
+		return reinterpret_cast<uint64_t>(p_next_pointer);
+	}
+}
+
+void OpenXRFbSpaceWarpExtensionWrapper::_on_instance_destroyed() {
+	cleanup();
+}
+
+void OpenXRFbSpaceWarpExtensionWrapper::_on_session_created(uint64_t p_instance) {
+	if (!fb_space_warp_ext) {
+		return;
+	}
+
+	ProjectSettings *project_settings = ProjectSettings::get_singleton();
+	bool is_project_setting_enabled = (bool)project_settings->get_setting_with_override("xr/openxr/extensions/application_space_warp");
+	if (!is_project_setting_enabled) {
+		fb_space_warp_ext = false;
+		return;
+	} else {
+		enabled = true;
+	}
+
+	String rendering_method = (String)project_settings->get_setting_with_override("rendering/renderer/rendering_method");
+	if (rendering_method == "forward_plus") {
+		UtilityFunctions::print_verbose("Disabling XR_FB_space_warp extension; this extension is not supported in the forward plus renderer");
+		fb_space_warp_ext = false;
+		enabled = false;
+		return;
+	}
+
+	get_openxr_api()->register_projection_views_extension(this);
+}
+
+void OpenXRFbSpaceWarpExtensionWrapper::_on_session_destroyed() {
+	if (!fb_space_warp_ext) {
+		return;
+	}
+
+	get_openxr_api()->unregister_projection_views_extension(this);
+	memdelete_arr(space_warp_info);
+}
+
+void OpenXRFbSpaceWarpExtensionWrapper::_on_state_ready() {
+	if (!fb_space_warp_ext || !enabled) {
+		return;
+	}
+
+	Ref<OpenXRInterface> openxr_interface = XRServer::get_singleton()->find_interface("OpenXR");
+	int view_count = openxr_interface->get_view_count();
+	int width = system_space_warp_properties.recommendedMotionVectorImageRectWidth;
+	int height = system_space_warp_properties.recommendedMotionVectorImageRectHeight;
+
+	String rendering_method = RenderingServer::get_singleton()->get_current_rendering_method();
```

Comment:
Checking rendering method seems wrong here - what we care about is the graphics API used by OpenXR. The "mobile" renderer could be using Vulkan, DirectX or Metal, and in each of those cases, we'd need to use different ids for the texture formats.

Didn't we add a method to get the name of the graphics API used by OpenXR specifically for this purpose?

Reply:
I've fixed this in my latest push by checking the render driver name here instead. I did end up removing the method returning the graphics API used by OpenXR since I deemed it not necessary, but I can always add it back in at some point if need be.